rdkPlugins/Logging/source/FileSink.cpp

```diff
@@ -0,0 +1,282 @@
+ * file. Does not attempt to seek the file descriptor back to the start
+ *
+ * If file limit is hit, will send data to /dev/null
+ *
+ * @param[in]   bufferFd    The fd to read from
+ */
+void FileSink::DumpLog(const int bufferFd)
+{
+    std::lock_guard<std::mutex> locker(mLock);
+
+    memset(mBuf, 0, sizeof(mBuf));
+
+    ssize_t ret;
+    ssize_t offset = 0;
+
+    // Read all the data from the provided file descriptor
+    while (true)
+    {
+        ret = read(bufferFd, mBuf, sizeof(mBuf));
+        if (ret <= 0)
+        {
+            break;
+        }
+
+        offset += ret;
+
+        if (offset <= mFileSizeLimit)
+        {
+            // Write to the output file
+            if (write(mOutputFileFd, mBuf, ret) < 0)
+            {
+                AI_LOG_SYS_ERROR(errno, "Write failed");
+            }
+        }
+        else
+        {
+            // Hit the limit, send the data into the void
+            if (!mLimitHit)
+            {
+                AI_LOG_WARN("Logger for container %s has hit maximum size of %zu",
+                            mContainerId.c_str(), mFileSizeLimit);
+            }
+            mLimitHit = true;
+            write(mDevNullFd, mBuf, ret);
+        }
+    }
+
+#if (AI_BUILD_TYPE == AI_DEBUG)
+    // Separate sections of log file for reabability
+    // (useful if we're writing lots of buffer dumps)
+    std::string marker = "---------------------------------------------\n";
+    write(mOutputFileFd, marker.c_str(), marker.length());
+#endif
+}
+
+/**
+ * @brief Called by the pollLoop when an event occurs on the container ptty.
+ *
+ * Reads the contents of the ptty and logs to a file
+ */
+void FileSink::process(const std::shared_ptr<AICommon::IPollLoop> &pollLoop, uint32_t events)
+{
+    std::lock_guard<std::mutex> locker(mLock);
+
+    if (events & EPOLLIN)
+    {
+        ssize_t ret;
+        ssize_t offset = 0;
+
+        memset(mBuf, 0, sizeof(mBuf));
+
+        while (true)
+        {
+            ret = TEMP_FAILURE_RETRY(read(mLoggingOptions.pttyFd, mBuf, sizeof(mBuf)));
+            if (ret <= 0)
+            {
+                // We've reached the end of the data we can read so we're done here
+                if (errno == EWOULDBLOCK)
+                {
+                    return;
+                }
+
+                // Something went wrong whilst reading
+                AI_LOG_SYS_ERROR(errno, "Read from container %s tty failed", mContainerId.c_str());
+                return;
+            }
+
+            offset += ret;
+            if (offset <= mFileSizeLimit)
+            {
+                // Write to the output file
+                if (write(mOutputFileFd, mBuf, ret) < 0)
+                {
+                    AI_LOG_SYS_ERROR(errno, "Write to %s failed", mOutputFilePath.c_str());
+                }
+            }
+            else
+            {
+                // Hit the limit, send the data into the void
+                if (!mLimitHit)
```

Comment:
You hit the limit and wrote to the dev/null one run before. I would get rid of this condition and move this message below to line 223. With current implementation user can lost exactly one last packet of logs and not get notified which can cause confusion.